### PR TITLE
Sebastian Herrera Hackerrank Collections Day 3

### PR DESCRIPTION
**Pull request by elrocetas**
Hackerrank exercises

### CODE REVIEW

File: SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/arraylist/ExerciseArraylist.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    public static final String ERROR_MESSAGE = "ERROR!";
+
+    public static void getArrayInfo() {
+
+        Scanner scanner = new Scanner(System.in);
+
+        int numberLines = scanner.nextInt();
+
+        ArrayList<ArrayList<Integer>> dynamicTable = new ArrayList<>();
```
Comment (luisod78) [inline SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/arraylist/ExerciseArraylist.java:20]: Is completely neccesary this "inception" ? ArrayList<ArrayList< ??
Also remember Code for Interface, not for Implementation:
https://www.javaguides.net/2018/06/java-collection-framework-best-practices.html

File: SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/hashset/ExerciseHashset.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        int testCases = scanner.nextInt();
+        String[] pairLeft = new String[testCases];
+        String[] pairRight = new String[testCases];
+
+        for (int index = 0; index < testCases; index++) {
+            pairLeft[index] = scanner.next();
+            pairRight[index] = scanner.next();
+        }
+
+        HashSet<String> pairs = new HashSet<>();
```
Comment (luisod78) [inline SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/hashset/ExerciseHashset.java:26]: remember Code for Interface, not for Implementation:
https://www.javaguides.net/2018/06/java-collection-framework-best-practices.html

File: SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/list/ExerciseList.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    public static void getListInfo() {
+        Scanner scanner = new Scanner(System.in);
+        int numElementsList = scanner.nextInt();
+        List<Integer> dataList = new ArrayList<>();
+        for (int index = 0; index < numElementsList; index++) {
+            dataList.add(scanner.nextInt());
+        }
+        int numberQueries = scanner.nextInt();
+        for (int index = 0; index < numberQueries; index++) {
```
Comment (luisod78) [inline SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/list/ExerciseList.java:24]: I think somebody is peeking at the code of their partners.....

**Review by luisod78 — CHANGES_REQUESTED** [review]

File: SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/list/ExerciseList.java (hunk 3 above)
Comment (elrocetas) [reply SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/list/ExerciseList.java:24]: Nooo,  I think great minds think alike. :woozy_face:

File: SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/arraylist/ExerciseArraylist.java (hunk 1 above)
Comment (elrocetas) [reply SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/arraylist/ExerciseArraylist.java:20]: Oh, okay. I think I have another solution

File: SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/ransom/ExerciseRansom.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+        Scanner scanner = new Scanner(System.in);
+        int magazineSize = scanner.nextInt();
+        int noteSize = scanner.nextInt();
+
+        scanner.nextLine();
+        String[] magazine = scanner.nextLine().split(" ");
+        String[] note = scanner.nextLine().split(" ");
+
+        HashMap<String, Integer> frequentWord = new HashMap<>();
```
Comment (luisod78) [inline SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/ransom/ExerciseRansom.java:25]: Also remember Code for Interface, not for Implementation:
https://www.javaguides.net/2018/06/java-collection-framework-best-practices.html

File: SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/arraylist/ExerciseArraylist.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+        Scanner scanner = new Scanner(System.in);
+
+        int numberLines = scanner.nextInt();
+
+        List<List> dynamicTable = new  ArrayList<>();
+
+        for (int tableIndex = 0; tableIndex < numberLines; tableIndex++) {
+            int arraySize = scanner.nextInt();
+            ArrayList<Integer> arrayElements = new ArrayList<>();
```
Comment (luisod78) [inline SebastianHerrera/bootcamp-java/src/main/java/hackerrank/day3/arraylist/ExerciseArraylist.java:25]: Also remember Code for Interface, not for Implementation:
https://www.javaguides.net/2018/06/java-collection-framework-best-practices.html

**Review by luisod78 — CHANGES_REQUESTED** [review]

**Review by luisod78 — APPROVED** [review]

**Review by luisalanmz — APPROVED** [review]
LGTM!

**Pull request merged**